src/_Classics_/knuth-morris-pratt/index.js

```diff
@@ -0,0 +1,100 @@
+
+/* *
+* The time complexity of KMP algorithm is O(n) in the worst case
+* Example use case: Pattern = ABCABCACA Text = AAABCBAABCABCACACABBCA
+* LPSArray = [ 0, 0, 0, 1, 2, 3, 4, 0, 1 ]
```

Comment:
It should be LPSArray = [ 0, 1, 0, 0, 1, 0 ]

Reply:
Hi @TheSTL the pattern had been updated to ABCABCACA. I feel this is a more intuitive example. I believe [ 0, 0, 0, 1, 2, 3, 4, 0, 1 ] to be the correct LPS, if there are any more queries or amendments you would like to be made, I’ll updage the PR as appropriate. Cheers.

Reply:
I think this example is more understandable from previous one to compute LPS.